### ISSUE TITLE
`#_oab_ask_library` not showing up during some requests

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1721]
**Describe the bug**

We're not showing the option to make an ILL in some configurations where I'd expect it (e.g when oa links are available, when subs are provided, etc).

This appears to have been happening (at least on dev) since Nov 2 / 3rd. I wasn't checking ghost inspector last week as I was away, so didn't catch it.

**To Reproduce**

See these tests

* https://app.ghostinspector.com/tests/5f23d623bfb7c24532041f92
* https://app.ghostinspector.com/tests/5f23d37a5e12c014aeae7dfb
* https://app.ghostinspector.com/tests/5f3d5e9591c8814c2084ad17

**Expected behavior**

In these cases, unless the config says otherwise, the option to make an ILL would be available

**Screenshots / GIF**

See tests

**Checklist (For internal use)**

* Take a breath. Bugs are okay.
* Stay clear of hangouts

Process checklist:
* Do we want a `fast` first, then a `good` fix? A `good` fix likely involves discussion, and may be slower. A fast fix means whoever is available fixes the problem as best they can immediately without too much consideration for the `good` fix.
- [ ] Discussion of solution. Remember, our intuitive fixes may be different (e.g UI centric, backend centric)

Post-fix
- [ ] How can we avoid bugs like this in future?

Likely we should discuss who's checking for ghost failures, how to make sure we remain responsive to them, etc.

- [X] How can we avoid regressions (e.g automated test)

**Comment by JosephMcArthur** [comment]
To help avoid this in future I've:

* I have improved some of the tests above to give a clearer failure if something like this happened
* I have fixed one test that has been erroring for a while so that once again failures can be trusted. We should expect green lights across the board again (if everything is in fact working)

**Comment by markmacgillivray** [comment]
I believe this is fixed, I tested and made a change to the backend find code. It appears that in some cases now where we have cached responses they are returning too fast for the system to complete the ILL lookup. I had already moved that out of the parallel loop to ensure it ran with all the metadata we could find, however the runner would still check to see if all metadata was already found and if so would just return. So now it will run the ILL check even if all metadata has already been found, which is what we want.

Fixing this on live means deploying backend code rather than frontend. I'll probably be doing that tomorrow anyway, but have a look at this on dev and let me know if you're happy with it.
